src/Microsoft.PowerShell.Commands.Utility/commands/utility/ShowMarkdownCommand.cs

```diff
@@ -80,21 +69,50 @@ protected override void ProcessRecord()
                     if (!string.IsNullOrEmpty(html))
                     {
                         string tmpFilePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString() + ".html");
-                        using (var writer = new StreamWriter(new FileStream(tmpFilePath, FileMode.Create, FileAccess.Write, FileShare.Write)))
+
+                        try
                         {
-                            writer.Write(html);
+                            using (var writer = new StreamWriter(new FileStream(tmpFilePath, FileMode.Create, FileAccess.Write, FileShare.Write)))
+                            {
+                                writer.Write(html);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            var errorRecord = new ErrorRecord(
+                                e,
+                                "ErrorWritingTempFile",
+                                ErrorCategory.WriteError,
+                                tmpFilePath);
+
+                            WriteError(errorRecord);
+                            return;
                         }
 
-                        if (outputBypassTestHook)
+                        if (InternalTestHooks.ShowMarkdownOutputBypass)
                         {
                             WriteObject(html);
                             return;
                         }
 
-                        ProcessStartInfo startInfo = new ProcessStartInfo();
-                        startInfo.FileName = tmpFilePath;
-                        startInfo.UseShellExecute = true;
-                        Process.Start(startInfo);
+                        try
+                        {
+                            ProcessStartInfo startInfo = new ProcessStartInfo();
+                            startInfo.FileName = tmpFilePath;
+                            startInfo.UseShellExecute = true;
+                            Process.Start(startInfo);
+                        }
+                        catch (Exception e)
+                        {
+                            var errorRecord = new ErrorRecord(
+                                e,
+                                "ErrorLaunchingDefaultApplication",
+                                ErrorCategory.InvalidOperation,
+                                targetObject : null);
```

Comment:
Make sense send `startInfo` argument?

Reply:
startInfo is internally generated. I think seeing it in the error will be more confusing than useful.